python/pyspark/context.py

```diff
@@ -152,6 +152,11 @@ def _do_init(self, master, appName, sparkHome, pyFiles, environment, batchSize,
         self.master = self._conf.get("spark.master")
         self.appName = self._conf.get("spark.app.name")
         self.sparkHome = self._conf.get("spark.home", None)
+
+        # Let YARN know it's a pyspark app, so it distributes needed libraries.
```

Comment:
Hmm, I kinda dislike having more coupling like this, but I can't see a better solution without making assumptions like "all code goes through spark-submit at some point", so I guess this will have to do.

Minor, but this should probably go into `_initialize_context` instead, since if there's already a `jsc` initialized, this code won't have any effect.